### PR TITLE
Fix #736, Section Reordering

### LINKED ISSUE #736: Proposed reordering of sections
Screenshot 2023-08-25 at 8 59 12 PM

Move all encoder-side parts into Annexes
The main part should consist mainly of normative bitstream and decoder-side items.

This should be done when most PRs have been merged to avoid merge conflicts.

### PR DESCRIPTION
**Pull request by sunghee-hwang**
***
<a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/iamf/pull/747.html" title="Last updated on Aug 31, 2023, 1:08 AM UTC (4f2435d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/iamf/747/287eada...4f2435d.html" title="Last updated on Aug 31, 2023, 1:08 AM UTC (4f2435d)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -913,11 +913,14 @@ When an [=Audio Element=] is composed of \(G(r)\) number of [=Audio Substream=]s
...
+Each [=Channel Group=] (or scalable channel audio layer) is associated with a different [=loudspeaker_layout=]. The IA decoder SHALL select one of the layers according to the following rules, in order:
 
 - The IA decoder SHOULD first attempt to select the layer with a [=loudspeaker_layout=] that matches the physical playback layout.
 - If there is no match, the IA decoder SHOULD select the layer with the closest [=loudspeaker_layout=] to the physical layout and then apply up- or down-mixing appropriately, after decoding and reconstruction of the channel audio. Sections [[#iamfgeneration-scalablechannelaudio-downmixmechanism]] and [[#processing-downmixmatrix]] provide examples of dynamic and static down-mixing matrices for some common layouts that MAY be used.
 
+The relationship among all [=Channel Group=]s for an [=Audio Element=] SHALL comply with [[#scalablechannelaudio-channelgroupformat]] and the relationship among all channel layouts indicated by [=loudspeaker_layout=]s specified in an [=Audio Element OBU=] SHALL comply with [[#scalablechannelaudio-channellayoutgenerationrule]].
+
```
Comment (sunghee-hwang) [inline index.bs:922]:
```suggestion
The relationship among all [=Channel Group=]s for the given scalable channel audio representation SHALL comply with [[#scalablechannelaudio-channelgroupformat]] and the relationship among all channel layouts indicated by [=loudspeaker_layout=]s specified in an [=Audio Element OBU=] SHALL comply with [[#scalablechannelaudio-channellayoutgenerationrule]].
```

**Review by felicialim — APPROVED** [review]

File: index.bs [hunk 2]
```diff
@@ -2591,7 +2623,77 @@ The 3.1.2ch down-mix matrix for 7.1.4ch is given below, where \(p = 0.707\).
...
+<tr>
+  <td>\(x^y\)</td><td>The value of \(x\) to the power of \(y\).</td>
+</tr>
+</table>
+
+## Q Format ## {#convention-qformat}
+
+<b>Qx.y</b>
+
+<b>Qx.y</b> indicates that it is stored as a (x+y+1)-bit, signed, two’s complement fixed-point value with y fractional bits. That is, a (x+y+1)-bit signed (two’s complement) integer, that is implicitly multiplied by the scaling factor 2^(−y).
```
Comment (felicialim) [inline index.bs:2692]:
```suggestion
<b>Qx.y</b> indicates that it is stored as a (x+y+1)-bit, signed, two’s complement fixed-point value with y fractional bits. That is, a (x+y+1)-bit signed (two’s complement) integer, that is implicitly multiplied by the scaling factor \(2^{−y}\).
```

File: index.bs [hunk 3]
```diff
@@ -1018,6 +1020,36 @@ Bit position : Channel Name
 
 <dfn noexport>output_gain</dfn> indicates the gain value to be applied to the mixed channels which are indicated by [=output_gain_flags=], where each mixed channel is generated by down-mixing two or more input channels. It is computed as \(20 \times \log_{10}(f)\), where \(f\) is the factor by which to scale the mixed channels. It is stored as a 16-bit, signed, two’s complement fixed-point value with 8 fractional bits (i.e., Q7.8)([[Q-Format]]).
 
+#### Channel Layout Generation Rule (Normative) #### {#scalablechannelaudio-channellayoutgenerationrule}
```
Comment (Frank-Ba) [inline index.bs:1023]: (Normative) can be removed here because it is in a normative section.

File: index.bs [hunk 4]
```diff
@@ -1018,6 +1020,36 @@ Bit position : Channel Name
...
+For a given channel layout (CL #n) of a channel-based input [=3D audio signal=], any list of CLs ({CL #i: i = 1, 2, ..., n}) for scalable channel audio SHALL conform with the following rules:
+- Xi ≤ Xi+1 and Yi ≤ Yi+1 and Zi ≤ Zi+1 except Xi = Xi+1, Yi = Yi+1 and Zi = Zi+1 for i = n-1, n-2, ..., 1, where the i-th channel layout CL #i = Xi.Yi.Zi, Xi is the number of surround channels, Yi is the number of LFE channels, and Zi is the number of height channels.
+- CL #i is one of the [=loudspeaker_layout=]s supported in this version of the specification.
+
+Scalable channel audio with [=num_layers=] > 1 SHALL only allow down-mix paths that conform to the rules above, as depicted in the figure below.
+
+<center><img src="images/Down-mix Path.png" style="width:90%; height:auto;"></center>
+<center><figcaption>IA Down-mix Path for scalable channel audio</figcaption></center>
+
+#### Channel Group Format (Normative) #### {#scalablechannelaudio-channelgroupformat}
```
Comment (Frank-Ba) [inline index.bs:1036]: (Normative) can be removed here because it is in a normative section.

**Review by Frank-Ba — APPROVED** [review]
Added minor suggested changes

File: index.bs [hunk 5]
```diff
@@ -1032,6 +1034,36 @@ Bit position : Channel Name
 
 <dfn noexport>output_gain</dfn> indicates the gain value to be applied to the mixed channels which are indicated by [=output_gain_flags=], where each mixed channel is generated by down-mixing two or more input channels. It is computed as \(20 \times \log_{10}(f)\), where \(f\) is the factor by which to scale the mixed channels. It is stored as a 16-bit, signed, two’s complement fixed-point value with 8 fractional bits (i.e., Q7.8)([[Q-Format]]).
 
+#### Channel Layout Generation Rule (Normative) #### {#scalablechannelaudio-channellayoutgenerationrule}
```
Comment (sunghee-hwang) [inline index.bs:1037]:
```suggestion
#### Channel Layout Generation Rule #### {#scalablechannelaudio-channellayoutgenerationrule}
```

File: index.bs [hunk 6]
```diff
@@ -1032,6 +1034,36 @@ Bit position : Channel Name
...
+For a given channel layout (CL #n) of a channel-based input [=3D audio signal=], any list of CLs ({CL #i: i = 1, 2, ..., n}) for scalable channel audio SHALL conform with the following rules:
+- Xi ≤ Xi+1 and Yi ≤ Yi+1 and Zi ≤ Zi+1 except Xi = Xi+1, Yi = Yi+1 and Zi = Zi+1 for i = n-1, n-2, ..., 1, where the i-th channel layout CL #i = Xi.Yi.Zi, Xi is the number of surround channels, Yi is the number of LFE channels, and Zi is the number of height channels.
+- CL #i is one of the [=loudspeaker_layout=]s supported in this version of the specification.
+
+Scalable channel audio with [=num_layers=] > 1 SHALL only allow down-mix paths that conform to the rules above, as depicted in the figure below.
+
+<center><img src="images/Down-mix Path.png" style="width:90%; height:auto;"></center>
+<center><figcaption>IA Down-mix Path for scalable channel audio</figcaption></center>
+
+#### Channel Group Format (Normative) #### {#scalablechannelaudio-channelgroupformat}
```
Comment (sunghee-hwang) [inline index.bs:1050]:
```suggestion
#### Channel Group Format #### {#scalablechannelaudio-channelgroupformat}
```

**Pull request merged**